plugins/catalog-react/src/hooks/useEntityListProvider.tsx

```diff
@@ -349,7 +397,15 @@ export const EntityListProvider = <EntityFilters extends DefaultEntityFilters>(
       prev: prevCursor ? () => setCursor(prevCursor) : undefined,
       next: nextCursor ? () => setCursor(nextCursor) : undefined,
     };
-  }, [enablePagination, outputState.pageInfo]);
+  }, [paginationMode, outputState.pageInfo]);
+
+  const setLimitCb = useCallback((newLimit: number) => {
+    setLimit(newLimit);
+  }, []);
+
+  const setOffsetCb = useCallback((newOffset: number) => {
+    setOffset(newOffset);
+  }, []);
```

Comment:
`useState` already is guaranteed to return static setters; this pair of useCallbacks will have no effect